src/components/valences-test/ValencesTest.test.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from "react";
+import { Router } from "react-router-dom";
+import { render, screen, within, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { createMemoryHistory } from "history";
+
+import ElementManager from "../../ElementManager";
+import ValencesTest from "./ValencesTest";
+
+const history = createMemoryHistory();
+
+const setup = () => {
+  ElementManager.loadElements();
+  return render(
+    <Router history={history}>
+      <ValencesTest />
+    </Router>
+  );
+};
+
+test("Should render a question and four answers", () => {
+  const { container } = setup();
+  const questionCard = container.querySelector(".question-card") as HTMLElement;
+  expect(questionCard).toBeInTheDocument();
+  const buttons = within(questionCard).getAllByRole("button");
+  expect(buttons).toHaveLength(4);
+});
+
+test("should display a new question when clicking on the correct answer", () => {
+  const { container } = setup();
+  const elementTag = container.querySelector(".element");
+  const currentElementSymbol = elementTag?.textContent!;
```

Comment:
is the element chosen randomly? if yes, we should document the fact here in the test, to explain why we our test is so complicated here ... if no, IMHO we should test by using `screen.getByText`

Reply:
Yes, the array of questions gets shuffled at `componentDidMount`  so what I am
01- getting the element that was displayed on the question card
02- getting the right answer from `ElementManger`
03- getting the button for the right answer
04- triggering user click event
05- testing if a new question was displayed